actions/transactions/update.go

```diff
@@ -11,30 +11,75 @@ import (
 
 // update will update a transaction
 // Update transaction godoc
-// @Summary		Update transaction
-// @Description	Update transaction
+// @Summary		Update transaction - Use (PATCH) /api/v1/transactions/{id} instead.
+// @Description	This endpoint has been deprecated. Use (PATCH) /api/v1/transactions/{id} instead.
 // @Tags		Transactions
 // @Produce		json
-// @Param		UpdateTransaction body UpdateTransaction true " "
+// @Param		UpdateTransaction body UpdateTransaction true "Pass update transaction request model in the body"
 // @Success		200 {object} models.Transaction "Updated transaction"
 // @Failure		400	"Bad request - Error while parsing UpdateTransaction from request body, tx not found or tx is not associated with the xpub"
 // @Failure 	500	"Internal Server Error - Error while updating transaction"
-// @Router		/v1/transaction [patch]
+// @DeprecatedRouter	/v1/transaction [patch]
 // @Security	x-auth-xpub
 func (a *Action) update(c *gin.Context) {
-	reqXPubID := c.GetString(auth.ParamXPubHashKey)
 
-	var requestBody UpdateTransaction
+	var requestBody OldUpdateTransaction
 	if err := c.Bind(&requestBody); err != nil {
 		spverrors.ErrorResponse(c, spverrors.ErrCannotBindRequest, a.Services.Logger)
 		return
 	}
+	id := requestBody.ID
+
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
+
+	// Get a transaction by ID
+	transaction, err := a.Services.SpvWalletEngine.UpdateTransactionMetadata(
+		c.Request.Context(),
+		reqXPubID,
+		id,
+		requestBody.Metadata,
+	)
+	if err != nil {
+		spverrors.ErrorResponse(c, err, a.Services.Logger)
+		return
+	} else if transaction == nil {
+		spverrors.ErrorResponse(c, spverrors.ErrCouldNotFindTransaction, a.Services.Logger)
+	} else if !transaction.IsXpubIDAssociated(reqXPubID) {
+		spverrors.ErrorResponse(c, spverrors.ErrAuthorization, a.Services.Logger)
+		return
+	}
+
+	contract := mappings.MapToOldTransactionContract(transaction)
+	c.JSON(http.StatusOK, contract)
+}
+
+// update will update a transaction
+// Update transaction godoc
+// @Summary		Update transaction
+// @Description	Update transaction
+// @Tags		Transactions
+// @Produce		json
+// @Param		UpdateTransactionRequest body UpdateTransactionRequest true "Pass update transaction request model in the body"
+// @Success		200 {object} response.Transaction "Updated transaction"
+// @Failure		400	"Bad request - Error while parsing UpdateTransaction from request body, tx not found or tx is not associated with the xpub"
+// @Failure 	500	"Internal Server Error - Error while updating transaction"
+// @Router		/api/v1/transactions/{id} [patch]
```

Comment:
I'm not sure about this - the JIRA ticket says `/api/v1/transactions/drafts/:id`, but there is a comment from Mariusz saying it should be `/api/v1/transactions/:id`

@dorzepowski ?

Reply:
👀

Reply:
It was mistake in the JIRA task, we thought that this endpoint allows for updating metadata of transaction draft, while in code it occurs that it's really working only for transactions.

Reply:
Okay, so I will resolve this comment.